tests/unit/alloc_test.cc

```diff
@@ -194,6 +196,37 @@ SEASTAR_TEST_CASE(test_realloc_nullptr) {
 
     return make_ready_future<>();
 }
+
+SEASTAR_TEST_CASE(test_bad_alloc_throws) {
+    // test that a large allocation throws bad_alloc
+    auto stats = seastar::memory::stats();
+
+    size_t size = stats.total_memory() * 2; // cannot be satisfied
```

Comment:
It can, in debug node, so this test should be conditional on the allocator used.

Reply:
@avikivity - it is already, since it is inside an `#ifndef SEASTAR_DEFAULT_ALLOCATOR` block shared by many of the tests, though I will adjust the comment to make it clearer.

I did also run all the tests in debug build and they pass.